### DISCUSSION TITLE
TailwindCSS and Vitepress

### DISCUSSION THREAD

**Discussion by danielmoessner** [vuejs/vitepress#582] [Q&A]
How can I enable Tailwindcss in Vitepress?

I've tried that https://github.com/vuejs/vitepress/issues/62 but it didn't work for me. I'm using the latest version of vitepress `0.22.3`.

My config looks like this:
```
import { defineConfig } from 'vitepress';


export default defineConfig({
    ...
    themeConfig: {
        sidebar: [
            {
                text: 'Components',
                collapsable: true,
                children: [
                    { text: 'ButtonNormal', link: '/components/button-normal' },
                    { text: 'CircleLoader', link: '/components/circle-loader' }
                ]
            }
        ]
    },
    theme: './theme/index.js',
})
```

and my theme/index.js looks like this:
```
import '../styles/index.postcss'

import DefaultTheme from "vitepress/dist/client/theme-default";
export default { ...DefaultTheme };
```

with index.postcss being like this:
```
@tailwind base;
@tailwind components;
@tailwind utilities;
```

I've got my custom components working but they use tailwindcss and I can't get the styling to work.

**Comment by CodyBontecou** [comment]
https://github.com/ky-is/vitepress-starter-tailwind is a good starter repo showcasing how to get this working.

**Comment by ndom91** [comment]
Reviving this issue as the linked example doesn't seem to work with the latest VitePress / Vite3 dependencies. 

I've copied almost exactly the same setup as in [ky-is/vitepress-starter-tailwind](https://github.com/ky-is/vitepress-starter-tailwind), except for that the `DefaultTheme` is now imported from `vitepress/theme` instead of the longer path used in the old(er) version.

- `vitepress@1.0.0-alpha.4`
- `vue@3.2.37`
- `tailwind@3.1.6`
- `vite@2.9.14`

**Reply by brc-dd** [reply]
Have a look at this one: https://stackblitz.com/edit/vite-uujwcv You can customize `.postcssrc` there according to your need. Also, VitePress now plays well with TailwindCSS's dark mode classes. If you face some conflicts between the styles of VP and those added by Tailwind, then consider using `layout: page`, or you can use this workaround: https://github.com/vuejs/vitepress/issues/199#issuecomment-1168325262

**Reply by ndom91** [reply]
That worked! Simply adding the `.postcssrc.json` with the tailwind plugin (and `postcss-preset-env`). Seemed to do the trick!

```json title=".postcssrc.json"
{
  "plugins": {
    "tailwindcss": {},
    "postcss-preset-env": {
      "autoprefixer": { "flexbox": "no-2009" },
      "features": { "custom-properties": false },
      "stage": 3
    }
  }
}
```

**Reply by rover1024** [reply]
Thanks, it works perfectly

**Comment by otabekoff** [comment]
OK, that's fine! But I don't think it's time to integrate Vitepress with other tools(UI libs). I think the focus of all developers should be on fixing issues and adding full functionality and creating a stable release.

So far this looks pretty good for you: https://github.com/ky-is/vitepress-starter-tailwind

**Reply by otabekoff** [reply]
Sorry if you misunderstood me. I used the word should. But in fact, all of the above is true. Thank you developers for your hard work until today. We appreciate your work. We look forward to more from the developers.

**Comment by indirectlylit** [comment]
Tailwind currently ignores all dot folders: https://github.com/tailwindlabs/tailwindcss/issues/8712
